### DISCUSSION TITLE
Composer support

### DISCUSSION THREAD

**Discussion by Mikubill** [Mikubill/sd-webui-controlnet#359] [Ideas]
Recently researchers from Alibaba Group/Ant Group released their paper "Composer: Creative and Controllable Image Synthesis with Composable Conditions" ([arXiv](https://arxiv.org/pdf/2302.09778.pdf)). However, their model requires at least 28GB VRAM to run, even with flash-attn and fp16 optimization (see https://github.com/damo-vilab/composer/issues/1). It may be better to consider adding some features proposed in that paper before they release "A light-weighted Latent-Composer built upon Stable Diffusion 2.1".

For upscale models: I think automatic1111 will handle this.

For conditioning modules:
1. Local conditioning module seems just as same as current multi-controlnet but without(?) deep encoding parts.
2. Global conditioning module looks similar to other xattn-aware method, which project and concat conditions to xattn hidden-state input and timestep embeddings (such as color histograms etc.).
![output-2983-2](https://user-images.githubusercontent.com/31246794/221335472-2257171c-c9cb-4333-97a0-bd478ddb5e96.png)

For Image Manipulation:
1. Variations: Create new images that are similar to a given image but vary in certain aspects by conditioning on a specific subset of its representations - maybe related to dynamic weighing of different input.
2. Interpolations: Traversing in the embedding space of global representations between two images - needs more explain.
3. Reconfigurations: obtain its latent by applying DDIM inversion conditioned on a set of its representations, then apply DDIM sampling starting from conditioned on a modified set of representations to obtain a variant of the image - same as Img2img alternative test in WebUI (https://arxiv.org/abs/2208.01626, https://github.com/bloc97/CrossAttentionControl, also discussed in https://github.com/Mikubill/sd-webui-controlnet/issues/249 and https://github.com/Mikubill/sd-webui-controlnet/issues/319), but needs some modification to make use of conditions.
4. Editable region: By conditioning Composer on a set of representations c along with a masked image m - Looks similar to masked inpainting with controlnet.

Ideas and suggestions welcome.

**Comment by catboxanon** [comment]
Would there still be benefits to gain from this for SD1.x models? Just curious since it looks to be an entirely new model with only tentative and possibly limited support for SD2.1. Sorry if this question should be an obvious answer.
